### ISSUE TITLE
False alert by health_chk.sk

### ISSUE THREAD

**Issue by gombaiviktor** [parodyne/esos#219]
Hi!

I'm using a Dell R720XD server with ESOS 1.3.7. In the server, there is a H310 (mini) Dell raid controller (lsi based). The controller is in pass through mode, so all of my disks presented in OS layer. It appears, that the reported disk status by the raid controller is JBOD in this mode. So, I receive a lot of alert like this:

** Warning! It appears a PERC physical drive (32:0) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:1) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:2) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:3) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:4) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:5) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:6) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:7) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:8) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:9) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:10) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a PERC physical drive (32:11) has failed on controller # 0!
** Physical drive state: JBOD
** Warning! It appears a MegaRAID physical drive on adapter 0 contains media errors!
** Physical media error count: 91

Cheers,
Viktor

**Comment by msmith626** [comment]
Can you try the attached health-check script and see if it resolves the issue with the 'JBOD' PD state?
[health_chk.sh.gz](https://github.com/parodyne/esos/files/2320881/health_chk.sh.gz)

**Comment by gombaiviktor** [comment]
Hello!

Sorry for the delay, I was on vacation.

I've tested the script and it works as expected (I'm aware of the media
errors, it's a test system). The output is the following:

[root@stor-3q1 ~]# sh ./health_chk.sh
Checking hardware RAID logical drives...

Checking hardware RAID physical drives...

Starting additional MegaRAID health checks...
Number of adapters found: 1
Checking physical drives for media errors...
** Warning! It appears a MegaRAID physical drive on adapter 0 contains
media errors!
** Physical media error count: 259
** Warning! It appears a MegaRAID physical drive on adapter 0 contains
media errors!
** Physical media error count: 1
Checked 12 physical drive(s).
Checking enclosure(s) health/state...

Checking MD RAID arrays...

It appears the '/usr/sbin/zpool' tool is not installed, or at least
is not executable. Skipping ZFS pool checks...

Physical RAM check...
Total Memory:           15592864 kB
Used Memory:            1034016 kB
Available Memory:       14558848 kB
Memory Used Percent:    7%

Disk (/ -> root tmpfs) space check...
Total Disk Space:       1536 MB
Used Disk Space:        641 MB
Avail. Disk Space:      896 MB
Disk Used Percent:      40%

Best Regards,
Viktor

Marc A. Smith <notifications@github.com> ezt írta (időpont: 2018. aug. 25.,
Szo, 19:12):
